tests/PHPStan/Type/Php/data/sprintf.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace PHPStan\Type\Php\Test;
+
+use function PHPStan\Testing\assertType;
+
+$foo = 'foo';
+assertType("'foo'", sprintf('%s', $foo));
+
+function foo (string $nonLiteral): void {
+	assertType('non-empty-string', sprintf('%s', $nonLiteral));
```

Comment:
When the format is just `%s` and we dont know whether $nonLiteral was a non-empty-string before, how can it be a non-empty-string after the sprintf-call?

Reply:
This is basically a test I added before I added my changes to make sure I wouldn't break anything. As far as I can tell, there were no tests for the return type of `sprintf()` before.
The return type of `non-empty-string` confused me to, but the current logic is incredibly simple there and this edge case of "the format is just placholder(s)" is probably not worth the effort of checking this.
I was thinking of writing a comment in the test, but now I think it's probably simpler to just add a string to the format and use `'foo %s'` or something along those lines.